ConwaysGameOfLife.Tests/GameControllerShould.cs

```diff
@@ -59,72 +59,87 @@ public void ReturnUniverseString_GivenInvalidInputsFollowedByValidInput(IEnumera
+            _controller.CreateInitialUniverse();
             _input.GetReadLine(input);
             var expected = input;
 
-            var result = _controller.CreateValidLocationString(gridWidth, gridLength);
+            var result = _controller.CreateValidLocationString();
 
             Assert.Equal(expected, result);
         }
 
         [Theory]
-        [InlineData(new string[] { "0,3" }, "2,2", "3,3", 3, 3)]
-        [InlineData(new string[] { "3,0" }, "0,0", "3,3" , 3, 3)]
-        [InlineData(new string[] { ",,", ";;" }, "2,2", "3,3", 3, 3)]
-        [InlineData(new string[] { "3,3,", "3,4;" }, "3,2", "4,3", 4, 3)]
-        [InlineData(new string[] { "3,3,3" },"3,2", "4,3", 4, 3)]
-        [InlineData(new string[] { "a,b" }, "1,1", "3,3", 3, 3)]
-        [InlineData(new string[] { "-1,4" }, "2,0", "3,3", 3, 3)]
-        [InlineData(new string[] { "4,-1" }, "2,2", "3,3", 3, 3)]
-        //[InlineData(new string[] { "4 4" }, "2,1", "4 3", 4, 3)] // ThrowOverflowOrFormatException
-        public void ReturnLocationString_GivenInvalidInputsFollowedByValidInput(IEnumerable<string> invalidInputs, string validInput, string universeInput, int gridWidth, int gridLength)
+        [InlineData(new string[] { "0,3" }, "2,2", "3,3")]
+        [InlineData(new string[] { "3,0" }, "0,0", "3,3")]
+        [InlineData(new string[] { ",,", ";;" }, "2,2", "10,10")] // failing
+        [InlineData(new string[] { "3,3,", "3,4;" }, "3,2", "4,4")] // failing
+        [InlineData(new string[] { "3,3,3" }, "3,2", "5,4")]
+        [InlineData(new string[] { "a,b", "0,a", "o,9" }, "1,1", "2,2")]
+        [InlineData(new string[] { "-1,4", "a,b", "0,a", "o,9" }, "2,0", "3,3")] // failing
+        [InlineData(new string[] { "4,-1" }, "2,2", "5,5")]
+        [InlineData(new string[] { "4 4" }, "2,1", "3,3")]
+        public void ReturnLocationString_GivenInvalidInputsFollowedByValidInput(IEnumerable<string> invalidInputs, string validInput, string universeInput)
         {
-            InputParser.ParseUniverse(universeInput);
+            _input.GetReadLine(universeInput);
+            _controller.CreateInitialUniverse();
             _input.GetReadLine(invalidInputs);
             _input.GetReadLine(validInput);
-            var expectedOutputLine1 = "Invalid input. Please try again.";
-            var expectedOutputLine2 = "Please enter the x, y coordinates for one live cell (e.g. '0,1').";
+            var invalidOutputMessageLine1 = "Invalid input. Please try again.";
+            var invalidOutputMessageLine2 = "Please enter the x, y coordinates for one live cell (e.g. '0,1').";
+            var count = invalidInputs.Count();
+
+            var result = _controller.CreateValidLocationString();
 
-            var result = _controller.CreateValidLocationString(gridWidth, gridLength);
-            var actualOutputLine1 = _output.GetWriteLine(0);
-            var actualOutputLine2 = _output.GetWriteLine(1);
+            for (var i = 1; i <= count; i += 2)

```

Comment:
same with this loop
```suggestion
            for (var i = 1; i <= count * 2; i += 2)
```

Reply:
ahh that's my bad, good pick up @Kedoodle ! I think at the moment this test is focusing too much on the implementation details and it would be enough to give it all the invalid inputs followed by the valid input, then check that the output is equal to the valid input. This way, it would inherently test that the method will stop expecting input only until it receives a valid output.